internal/cluster/validations/validations.go

```diff
@@ -72,11 +76,25 @@ const (
 )
 */
 
-func ValidatePullSecret(secret string) error {
-	_, err := ParsePullSecret(secret)
+func ValidatePullSecret(secret string, username string, authHandler auth.AuthHandler) error {
+	creds, err := ParsePullSecret(secret)
 	if err != nil {
 		return err
 	}
+
+	if authHandler.EnableAuth {
+		r, ok := creds["cloud.openshift.com"]
+		if !ok {
+			return fmt.Errorf("Pull secret does not contain auth for cloud.openshift.com")
+		}
+		user, err := authHandler.AuthAgentAuth(r.AuthRaw)
+		if err != nil {
+			return fmt.Errorf("Failed to authenticate Pull Secret Token")
+		}
+		if (user.(*ocm.AuthPayload)).Username != username {
```

Comment:
We don't intent to support creation of cluster for other user?

Reply:
Not for now. The Authz part will not allow to see other user clusters.
@avishayt please correct me if I am wrong.

See additional discussion in JIRA MGMT-1814